spec/system/country_page_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe "Country page", type: :system, driver: :selenium_chrome, js: true do
+  include_context 'countries_setup'
+
+  let(:country_path) { country_path = GeoEntity.last.name.downcase.gsub(/ /, '-') }
+
+  before do
+    allow(GeoEntity).to receive(:permitted_countries).and_return(GeoEntity.all)
+    country
+    many_species
+  
+    visit "/#{country_path}"
+  end
+
+  scenario 'Visiting a country page' do
+    expect(page).to have_content(GeoEntity.last.actual_name)
+  end
+
+  scenario 'Finding the total length of the coastline' do
+    chart_title = find('.habitat-representation__chart-title')
+
+    unrounded_coastline = GeoEntity.last.coastal_stat.total_coast_length
+    coastline_length = ActiveSupport::NumberHelper.number_to_delimited(unrounded_coastline.round)
+
+    expect(chart_title.text).to eq("Percentage of coastline (#{coastline_length} km) covered by habitats")
+  end
+
+  scenario 'Looking at example of species section' do
+    expect(page).to have_content('Total number of species: 20')
+
+    species_common_names = GeoEntity.last.species.take(3).pluck(:common_name)
+
+    expect(page).to have_content(species_common_names[0])
+    expect(page).to have_content(species_common_names[1])
+    expect(page).to have_content(species_common_names[2])
```

Comment:
Perhaps these should target the elements where you expect that content to be - or at least limit it to the component container. That would be more robust, in case in the future, species names appeared elsewhere on the page. Just a small thing.